oca_port/utils/misc.py

```diff
@@ -62,3 +62,40 @@ def _render_output(self, output, data):
     def _render_output_json(self, data):
         """Render the data as JSON."""
         return json.dumps(data)
+
+
+class SmartDict(dict):
+    """Dotted notation dict."""
+
+    def __getattr__(self, attrib):
+        val = self.get(attrib)
+        return self.__class__(val) if type(val) is dict else val
+
+
+REF_REGEX = r"((?P<org>[\w-]+)/)?((?P<repo>[\w-]+)#)?(?P<branch>.*)"
+
+
+def parse_gh_ref(ref):
+    """Parse github reference in the form [org/][repo#]branch"""
+    group = re.match(REF_REGEX, ref)
+    return SmartDict(group.groupdict()) if group else None
+
+
+def make_gh_info(kind, ref, remote=None):
+    info = parse_gh_ref(ref)
+    if not info:
+        # FIXME
+        raise ValueError(f"No valid {kind}")
+    info["_kind"] = kind
+    info["remote"] = remote or info.org
+    if not info.org:
+        info["org"] = "OCA"
+    return info
```

Comment:
All this part is not yet used, are you sure you want to commit them now?

Reply:
yes